runtime/error.c

```diff
@@ -1,12 +1,103 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+
 #include "balrt.h"
+#include "third-party/libbacktrace/backtrace.h"
+
+#define INITIAL_PC_COUNT 1
+#define MAX_PC_COUNT 256
+#define SKIP_FROM_END 1 // This number counts from the last stack frame
+#define THREAD 0
+
+struct backtrace_state *state;
+
+typedef struct {
+    uint32_t npcs;
+    uint32_t szpcs;
+    PC *pcs;
+} Trace;
+
+static void onError(void *data, const char *msg, int errnum) {
+    fprintf(stdout, "Error : libbacktrace : %s\n", msg);
+    fflush(stdout);
+}
+
+static int onFrame(void *data, PC pc, const char *filename, int lineno, const char *function) {
+    if (function == NULL || filename == NULL)
+        return 0;
+
+    fprintf(stdout, "%s", function);
+    fprintf(stdout, " %s", filename);
+    fputs(":", stdout);
+    fprintf(stdout, "%" PRIi32, lineno);
+    putc('\n', stdout);
+    fflush(stdout);
+    return 0;
+}
+
+static int onPC(void *data, PC pc) {
+    Trace *trace = (Trace *)data;
+    uint32_t npcs = trace->npcs;
+    uint32_t szpcs = trace->szpcs;
+    if (unlikely(npcs == MAX_PC_COUNT))
+        return 1;
+    if (npcs == szpcs) {
+        szpcs = szpcs << 1;
+        trace->szpcs = szpcs;
+        void *p = realloc(trace->pcs, sizeof(PC) * szpcs);
+        if (p == 0)
```

Comment:
Brace style in this project is to use braces even when there is a single statement.

Reply:
Changed